K8SPG-512 Documented how to change the primary (#164)
percona/k8spg-docs: docs/change-primary.md

# Change the PostgreSQL primary instance

The Operator uses PostgreSQL high-availability implementation based on the [Patroni template :octicons-link-external-16:](https://patroni.readthedocs.io/en/latest/faq.html#concepts-and-requirements).
This means that each PostgreSQL cluster includes one member availiable for read/write transactions (PostgreSQL primary instance, or leader in terms of Patroni) and a number of replicas which can serve read requests only (standby members of the cluster).

> You may wish to manually change the primary instance in your PostgreSQL cluster to achieve more control and meet specific requirements in various scenarios like planned maintenance, testing failover procedures, load balancing and performance optimization activities and the like.

Reviewer: ```suggestion
You may wish to manually change the primary instance in your PostgreSQL cluster to achieve more control and meet specific requirements in various scenarios like planned maintenance, testing failover procedures, load balancing and performance optimization activities.
```

Primary instance is re-elected during the automatic failover (Patroni's "leader race" mechanism), but still there are use cases to controll this process manually.

> In Percona Operator, the primary instance change can be controlled by the `patroni.switchover` section of the `deploy/cr.yaml` manifest. It allows you to enable switchover targeting a specific PostgreSQL instance as the new primary, or just running a failover if PostgreSQL cluster has entered a bad state.

Reviewer: ```suggestion
In Percona Operator, the primary instance election can be controlled by the `patroni.switchover` section of the Custom Resource manifest. It allows you to enable switchover targeting a specific PostgreSQL instance as the new primary, or just running a failover if PostgreSQL cluster has entered a bad state.
```

This document provides instructions how to change the primary instance manually. 

For the following steps, we assume that you have the PostgreSQL cluster up and running. The cluster name is `cluster1`. 

> 1. Check the information about the *cluster instances*. Cluster instances (actually, specifically configured parts of your PostgreSQL cluster) are defined in the `spec.instances` Custom Resource section. By default you have one cluster instance named `instance1` with 3 PostgreSQL instances in it.

Reviewer: ```suggestion
1. Check the information about the *cluster instances*. Cluster instances are defined in the `spec.instances` Custom Resource section. By default you have one cluster instance named `instance1` with 3 PostgreSQL instances in it.
```

This means that each PostgreSQL cluster includes one member availiable for read/write transactions (PostgreSQL primary instance, or leader in terms of Patroni) and a number of replicas which can serve read requests only (standby members of the cluster).

You may wish to manually change the primary instance in your PostgreSQL cluster to achieve more control and meet specific requirements in various scenarios like planned maintenance, testing failover procedures, load balancing and performance optimization activities.
Primary instance is re-elected during the automatic failover (Patroni's "leader race" mechanism), but still there are use cases to controll this process manually.

In Percona Operator, the primary instance election can be controlled by the `patroni.switchover` section of the Custom Resource manifest. It allows you to enable switchover targeting a specific PostgreSQL instance as the new primary, or just running a failover if PostgreSQL cluster has entered a bad state.

This document provides instructions how to change the primary instance manually. 

For the following steps, we assume that you have the PostgreSQL cluster up and running. The cluster name is `cluster1`. 

> 1. Check the information about the [cluster instances](operator.md#operator-instances-section). Cluster instances are defined in the `spec.instances` Custom Resource section. By default you have one cluster instance named `instance1` with 3 PostgreSQL instances in it.

Reviewer 2: By default you have one cluster instance named `instance1` with 3 PostgreSQL instances in it.
 
 It's a bit unclear. one instance with 3 instances?
